README.md

```diff
@@ -44,24 +43,31 @@ Data Sources take two type parameters:
 </ol>
 
 ```scala
+import cats.data.NonEmptyList
+
 trait DataSource[Identity, Result]{
-  def fetch(ids: NonEmptyList[Identity]): Eval[Map[Identity, Result]]
+  def fetchOne(id: Identity): Query[Option[Result]]
+  def fetchMany(ids: NonEmptyList[Identity]): Query[Map[Identity, Result]]
 }
 ```
 
 We'll implement a dummy data source that can convert integers to strings. For convenience, we define a `fetchString` function that lifts identities (`Int` in our dummy data source) to a `Fetch`. 
 
 ```scala
-import cats.Eval
 import cats.data.NonEmptyList
 import cats.std.list._
-
 import fetch._
 
 implicit object ToStringSource extends DataSource[Int, String]{
-  override def fetch(ids: NonEmptyList[Int]): Eval[Map[Int, String]] = {
-    Eval.later({
-      println(s"ToStringSource $ids")
+  override def fetchOne(id: Int): Query[Option[String]] = {
+    Query.later({
```

Comment:
Definitely, i think i'm going to simplify the way of constructing queries. I go into detail about this in the docs but `Query#sync` and `Query#async` should be mentioned in the README.